From e6c39c2219f9d308ea57c09fc1b0b5b151b52b7d Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Mon, 16 Apr 2018 11:33:11 +0300
Subject: [PATCH] bus: fsl-mc: define macros for iommu enablement for fsl-mc
 bus

Macros to get coherency and the container device of the devices on
fsl-mc bus are required to suport SMMU for this bus. This patch
defines the same.

Signed-off-by: Nipun Gupta <nipun.gupta@nxp.com>
Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
---
 include/linux/fsl/mc.h | 4 ++++
 1 file changed, 4 insertions(+)

--- a/include/linux/fsl/mc.h
+++ b/include/linux/fsl/mc.h
@@ -356,6 +356,10 @@ int mc_send_command(struct fsl_mc_io *mc
 #define fsl_mc_cont_dev(_dev) (fsl_mc_is_cont_dev(_dev) ? \
 	(_dev) : (_dev)->parent)
 
+#define fsl_mc_is_dev_coherent(_dev) \
+	(!((to_fsl_mc_device(_dev))->obj_desc.flags & \
+	FSL_MC_OBJ_FLAG_NO_MEM_SHAREABILITY))
+
 /*
  * module_fsl_mc_driver() - Helper macro for drivers that don't do
  * anything special in module init/exit.  This eliminates a lot of
